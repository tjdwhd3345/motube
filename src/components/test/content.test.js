import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Content from "../content/content";

// props으로 selectedVideo가 주어져야 함. mocking 필요
const fakeSelectedVideo = {
  id: 1,
  snippet: {
    title: "test-title",
    publishedAt: "2022-12-20",
    channelTitle: "test-channel-title",
    description: "test-desc",
  },
  statistics: {
    viewCount: 99_999_999,
  },
  channelThumbnail: {
    url: "test-url",
  },
};

describe("Content", () => {
  it("renders correctly", () => {
    const selectedVideo = {
      ...fakeSelectedVideo,
      statistics: { ...fakeSelectedVideo.statistics, likeCount: 10_000_000, dislikeCount: 3_333 },
    };
    const { asFragment } = render(<Content selectedVideo={selectedVideo} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("iframe 속성 확인", () => {
    render(<Content selectedVideo={fakeSelectedVideo} />);

    const iframe = screen.getByTestId("iframe");
    expect(iframe.src).toBe(`https://www.youtube.com/embed/${fakeSelectedVideo.id}?autoplay=1`);
  });

  it("like, dislike가 없으면 좋아요, 싫어요 문구 표시", () => {
    render(<Content selectedVideo={fakeSelectedVideo} />);

    expect(screen.getByText("좋아요")).toBeInTheDocument();
    expect(screen.getByText("싫어요")).toBeInTheDocument();
  });

  it("영상 설명 더보기, 간략히 toggle 확인", async () => {
    render(<Content selectedVideo={fakeSelectedVideo} />);

    const button = screen.getByRole("button");
    userEvent.click(button);

    expect(screen.getByText("간략히")).toBeInTheDocument();
  });
});
