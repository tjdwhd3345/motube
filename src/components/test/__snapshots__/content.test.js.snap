// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
<DocumentFragment>
  <section
    class="detail"
  >
    <iframe
      allowfullscreen=""
      class="video"
      data-testid="iframe"
      frameborder="0"
      id="ytplayer"
      src="https://www.youtube.com/embed/1?autoplay=1"
      title="motube video player"
    />
  </section>
  <h2
    class="title"
  >
    test-title
  </h2>
  <div
    class="firstInfo"
  >
    <div>
      <span
        class="playCounts"
      >
        조회수 99,999,999 회
      </span>
      <span
        class="publishedDate"
      >
        2022.12.20
      </span>
    </div>
    <div
      class="thumbs"
    >
      <i
        class="fas fa-thumbs-up"
      />
      <span
        class="likeThumb"
      >
        10,000,000
      </span>
      <i
        class="fas fa-thumbs-down"
      />
      <span>
        3,333
      </span>
    </div>
  </div>
  <div
    class="secondInfo"
  >
    <div
      class="channel"
    >
      <img
        alt="channelThumbnail"
        class="channelThumbnail"
        src="test-url"
      />
      <span
        class="channelTitle"
      >
        test-channel-title
      </span>
    </div>
    <div
      class="description "
    >
      test-desc
    </div>
    <button
      class="button"
    >
      더보기
    </button>
  </div>
</DocumentFragment>
`;
